transcom/mymove
# B21576 int add order types for sc too

## Agility ticket
[B21576](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-21576)

## Summary
In B21575, 2 new order types were added for the customer:  Student Travel, Early Return of Dependents.  B21576 adds these new order types for SC and TOO.  When the SC or TOO edits orders, they will be able to select the new order types and "has dependents" will update to "yes" on the backend.  In addition, the new order types will display in the "Orders" section of the "Move Details Page" for the SC or TOO.  And, finally, when SC creates a move, the "Orders" page will behave exactly like the customer version of this page.

### How to test
In the .envrc.local file, add the following to turn on the feature flag:
export FEATURE_FLAG_ENABLE_ALASKA=true
restart server and client

=====
Create a move with a HHG shipment and submit it.
* Login as customer
* On the Orders page,
  - Set order type to: permanent change of station
  - Set has dependents to: no.
* Note the move code.
* Keep the customer page open.  The orders section will be viewed to confirm changes.
* Open a browser window as SC
* Open a browser window as TOO

=====
Test:  SC: edit orders, Student Travel
* Login as SC
* Open the move that was just created
* Order type: Student Travel
  - Edit the orders
  - Select order type Student Travel
  - Save changes
  - In the "Orders" section, verify "Orders Type" is "Student Travel"
  - On the customer page,
    * Go to the "Orders" section: move main page > review your request > orders
    * Verify
      - Orders type is "Student Travel"
      - Dependents is "Yes"
    * Reset Orders
      - Edit orders
        * Set "Orders type" to "permanent change of station"
        * Set "dependents" to "no"
    * Click button "return home"
  - On TOO page
    * Open the move using search and the move code
    * Go to the "Move Details" tab
    * Confirm "orders type" is "Student Travel" in the Orders section.
    * Close the move

=====
Test:  SC: edit orders, Early Return of Dependents
* Login as SC
* Open the move that was just created
* Order type: Early Return of Dependents
  - Edit the orders
  - Select order type Early Return of Dependents
  - Save changes
  - In the "Orders" section, verify "Orders Type" is "Early Return of Dependents"
  - On the customer page,
    * Go to the "Orders" section: move main page > review your request > orders
    * Verify
      - Orders type is "Early Return of Dependents"
      - Dependents is "Yes"
    * Reset Orders
      - Edit orders
        * Set "Orders type" to "permanent change of station"
        * Set "dependents" to "no"
    * Click button "return home"
  - On TOO page
    * Open the move using search and the move code
    * Go to the "Move Details" tab
    * Confirm "orders type" is "Early Return of Dependents" in the Orders section.
    * Close the move

=====
Test:  SC: edit orders, Select neither Early Return of Dependents nor Student Travel
* Login as SC
* Open the move that was just created
* Order type: Local move
  - Edit the orders
  - Select order type Local move
  - Save changes
  - In the "Orders" section, verify "Orders Type" is "Local move"
  - On the customer page,
    * Go to the "Orders" section: move main page > review your request > orders
    * Verify
      - Orders type is "Local move"
      - Dependents is "No"
    * Reset Orders
      - Edit orders
        * Set "Orders type" to "permanent change of station"
        * Set "dependents" to "no"
    * Click button "return home"
  - On TOO page
    * Open the move using search and the move code
    * Go to the "Move Details" tab
    * Confirm "orders type" is "Local move" in the Orders section.
    * Close the move

=====
Test:  SC: create move
* login as SC
* search for customer using name "Spaceman".
* select a customer and click "create move".
* at the move orders page
  - set initial values all fields:
    - order type: Permanent Change of Station
    - remaining fields: any value is fine
* verification
  - verify: order type,  changing existing type to another existing type
    - change order type to local move
    - verify no fields are changed
  - verify: order type, changing existing type to new type (student travel)
    - change order type to student travel
    - verify "are dependents included" defaults to "yes"
    - verify "are dependents included" is disabled
  - verify: order type, changing new type (student travel) to existing type
    - change order type to permanent change of station
    - verify "are dependents included" defaults to no selection
    - verify "are dependents included" is enabled
  - verify: order type, changing existing type to new type (early return of dependents)
    - change order type to student travel
    - verify "are dependents included" defaults to "yes"
    - verify "are dependents included" is disabled
  - verify: order type, changing new type (early return of dependents) to existing type
    - change order type to permanent change of station
    - verify "are dependents included" defaults to no selection
    - verify "are dependents included" is enabled
  - verify: order type, changing new type (early return of dependents) to to new type (student travel)
    - change order type to early return of dependents
    - verify "are dependents included" defaults to "yes"
    - verify "are dependents included" is disabled
    - change order type to student travel
    - verify "are dependents included" defaults to "yes"
    - verify "are dependents included" is disabled

=====
Test: turn off feature flag
In the .envrc.local file, add the following to turn on the feature flag:
export FEATURE_FLAG_ENABLE_ALASKA=false
restart server and client
* SC, Create a move: On the orders page, verify neither Student Travel or Early Return of Dependents shows up in the drop down list
* SC, Edit orders: On the orders page, verify neither Student Travel or Early Return of Dependents shows up in the drop down list

## ANDI Screenshots
SC: move, edit orders
![image](https://github.com/user-attachments/assets/40e48a70-e141-4e38-83f8-55626f5ef7ce)

SC: move, orders section
![image](https://github.com/user-attachments/assets/31417246-2dd3-4945-9652-93956c428140)

TOO: move, orders section
![image](https://github.com/user-attachments/assets/bac8ac10-e9ee-4717-ae46-2a4c262870ab)

SC: create move
![image](https://github.com/user-attachments/assets/c0b72351-a810-4dcf-866e-1eb158555aef)
